app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -99,6 +99,7 @@ def self.find_authenticator(name)
     end
 
     Discourse.auth_providers.each do |provider|
+      raise Discourse::InvalidAccess.new("provider is not enabled") unless provider.enabled_setting.nil? || SiteSetting.send(provider.enabled_setting)
```

Comment:
We should probably translate the string here.

Reply:
A nitpick here is that if the line is more than 80 characters long, we prefer to write the `if/unless` statement on multiple lines.